### PR TITLE
[12.0][ADD] module sale_subscription

### PR DESCRIPTION
**Pull request by i-vyshnevska**
[BSASP-63](https://jira.camptocamp.com/browse/BSASP-63)

### CODE REVIEW

File: sale_subcription_sections/tests/test_subscription_sections.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from odoo.addons.sale_subscription.tests.common_sale_subscription import TestSubscriptionCommon
+from odoo.tests import tagged
+
+@tagged('post_install', '-at_install')
+class TestSubscriptionSections(TestSubscriptionCommon):
+
+    def test_sale_order(self):
+
+        subscription = self.subscription
+        subscription.write({
```
Comment (simahawk) [inline sale_subcription_sections/tests/test_subscription_sections.py]: use tracking_disable as I doubt the base class does it

File: sale_subcription_sections/tests/test_subscription_sections.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            (1, 'TestProduct'),
+            (2, 'Subscription'),
+            (3, 'TestProduct3'),
+            (5, 'Note'),
+            (10, 'TestProduct2'),
+        ]
+
+        res = [
+            (line.sequence, line.name)
+            for line in invoice.mapped('invoice_line_ids')
```
Comment (simahawk) [inline sale_subcription_sections/tests/test_subscription_sections.py]: FTR you can do this like `invoice.mapped(lambda x: (x.sequence, x.name))`

File: sale_subcription_sections/views/sale_subscription.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <record id="sale_subscription_line_subscription_ids" model="ir.ui.view">
+        <field name="name">sale.subscription.line section</field>
+        <field name="model">sale.subscription</field>
+        <field name="inherit_id" ref="sale_subscription.sale_subscription_view_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='recurring_invoice_line_ids']" position="attributes">
```
Comment (simahawk) [inline sale_subcription_sections/views/sale_subscription.xml]: Curiosity: where's this widget?

Comment (i-vyshnevska) [reply sale_subcription_sections/views/sale_subscription.xml]: in accounting https://github.com/odoo/odoo/blob/12.0/addons/account/static/src/js/section_and_note_fields_backend.js

**Pull request merged**